src/Integration.Vsix/CFamily/FileConfig.cs

```diff
@@ -247,32 +251,41 @@ public Request ToRequest(string path)
                 return request;
             }
 
-            internal /* for testing */ static string ConvertPlatformName(string platformName)
+            internal /* for testing */ static bool isPlatformX64(string platformName)
             {
                 switch (platformName)
                 {
                     default:
                         throw new ArgumentException($"Unsupported PlatformName: {platformName}", nameof(platformName));
                     case "Win32":
-                        return "x86";
+                        return false;
                     case "x64":
-                        return "x64";
+                        return true;
                 }
             }
 
-            internal /* for testing */ static string ConvertPlatformToolset(string platformToolset)
+            internal /* for testing */ static string getCompilerVersion(string platformToolset, string vcToolsVersion)
             {
                 switch (platformToolset)
                 {
                     default:
                         throw new ArgumentException($"Unsupported PlatformToolset: {platformToolset}", nameof(platformToolset));
                     case "": // Bug https://github.com/SonarSource/sonarlint-visualstudio/issues/804
                         throw new ArgumentException(Strings.Daemon_PlatformToolsetNotSpecified);
+                    // VCToolsVersion is only avaialble from VS2017 onward
+                    // Before VS2017 the platform was enough to deduce the compiler version
                     case "v142":
-                        return "19.20.00";
                     case "v141":
                     case "v141_xp":
-                        return "19.10.00";
+                        int index = vcToolsVersion.IndexOf('.');
```

Comment:
@abbas-sabra-sonarsource I tested this on VS2015 -- the platform toolset is V140 and lower, so this code is not called. 
Would there be any way to have a toolset higher than that? To make sure that we're not getting a null exception.

Reply:
@rita-g-sonarsource I confirm the new code will not be hit on VS2015. That is expected because there is only 19.00.00 for v140 and v140_xp.
No, it isn't possible to set a higher toolset.